src/fstab-generator/fstab-generator.c

```diff
@@ -962,6 +1118,8 @@ int main(int argc, char *argv[]) {
                 }
         }
 
+        r = add_mounts_from_cmdline();
```

Comment:
I think the error handling is incorrect here. Main error code returned by generator depends on value of `r`. Previously, generator would return error return code when `fstab` processing failed. Now, (w/o any units on the command line which is the default case) this potential error would be obscured by `0` returned from `add_mounts_from_cmdline()` where there are no mounts on the command line.